models/junos/l2_interfaces/junos_l2_interfaces.yaml

```diff
@@ -0,0 +1,83 @@
+---
+GENERATOR_VERSION: '1.0'
+
+ANSIBLE_METADATA: |
+    {
+      'metadata_version': '1.1',
+      'status': ['preview'],
+      'supported_by': 'network'
+    }
+NETWORK_OS: junos
+RESOURCE: l2_interfaces
+COPYRIGHT: Copyright 2019 Red Hat
+
+DOCUMENTATION: |
+  module: junos_l2_interfaces
+  version_added: 2.9
+  short_description: Manage Layer-2 interface on Juniper JUNOS devices.
+  description: This module provides declarative management of Layer-2 interface on Juniper JUNOS devices.
+  author: Ganesh Nalawade (@ganeshrn)
+  options:
+    config:
+      description: A dictionary of Layer-2 interface options
+      type: list
+      elements: dict
+      suboptions:
+        name:
+          description:
+          - Full name of interface, e.g. ge-0/0/1.
+          type: str
+          required: True
```

Comment:
Do you really think that this should be required? IMHO we can just make this optional.

Reply:
nvm, I misread this as interface's description!